packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts

```diff
@@ -152,12 +154,14 @@ export class PluginLedgerConnectorEthereum
 
   public async shutdown(): Promise<void> {
     this.log.info(`Shutting down ${this.className}...`);
-    const provider = this.web3.currentProvider;
-    if (provider && typeof provider == "object") {
-      if ("disconnect" in provider) {
-        provider.disconnect(1000, "shutdown");
-      }
-    }
+    // // NOTE: This causes `verifier-integration-with-ethereum-connector.test.ts` to crash (some kind of double shutdown?)
+    // // TODO: Investigate further, for now seem to work without it.
```

Comment:
@outSH The API server calls the shutdown methods on plugins during its own shutdown procedure. Is it possible that your test case is calling shutdown both on the API server object and directly on the plugin as well (which would result in the described behavior)

[1] https://github.com/hyperledger/cacti/blob/main/packages/cactus-cmd-api-server/src/main/typescript/api-server.ts#L432

[1] 
```typescript
public async shutdown(): Promise<void> {
  this.log.info(`Shutting down API server ...`);

  const registry = await this.getOrInitPluginRegistry();

  const webServicesShutdown = registry
    .getPlugins()
    .filter((pluginInstance) => isIPluginWebService(pluginInstance))
    .map((pluginInstance: ICactusPlugin) => {
      return (pluginInstance as IPluginWebService).shutdown();
    });
    // ... etc.
 }

```

Reply:
@petermetz Thanks for the hint, unfortunately, I'm doing only shutdown through api server so it's not it :/